statement.go

```diff
@@ -470,6 +469,77 @@ func (s *Stmt) ExecContext(ctx context.Context, nargs []driver.NamedValue) (driv
 	return &result{ra}, nil
 }
 
+// ColumnCount returns the number of columns that will be returned by executing the prepared statement.
+// If any of the column types is invalid (which can happen when the type is ambiguous), the result will be 1.
+// Returns an error if the statement is closed or uninitialized.
+func (s *Stmt) ColumnCount() (int, error) {
```

Comment:
Should we return `uint` here? Then, we can also take `uint` in the below functions and can skip the `n < 0` check?